cypress/integration/dib-travel/company-settings/travel-policy.spec.ts

```diff
@@ -0,0 +1,197 @@
+    cy.get('.cdk-overlay-container dib-travel-policy-dialog .dib-select').select(
+      travelPolicyForm.travelPolicyTypeTrain
+    );
+    cy.get('.cdk-overlay-container dib-travel-policy-dialog input[name=name]').type(travelPolicyForm.travelPolicyName);
+    cy.get('.cdk-overlay-container dib-travel-policy-dialog input[name=numberOfDaysInAdvance]').type(
+      travelPolicyForm.numberOfDaysInAdvance
+    );
+    cy.get('.cdk-overlay-container dib-travel-policy-dialog dib-checkbox').click({ multiple: true });
+    cy.get('.cdk-overlay-container dib-travel-policy-dialog input[name=budget]').type(
+      travelPolicyForm.travelPolicyBudget
+    );
+    cy.get('.cdk-overlay-container dib-travel-policy-dialog .add-btn').click();
+    cy.get('.cdk-overlay-container dib-travel-policy-dialog dib-list-item input[placeholder=from]').type(
+      travelPolicyForm.trainFrom
+    );
+    cy.get('.cdk-overlay-container .cdk-overlay-pane').contains(travelPolicyForm.trainFrom).click();
+    cy.get('.cdk-overlay-container dib-travel-policy-dialog dib-list-item input[placeholder=to]').type(
+      travelPolicyForm.trainTo
+    );
+    cy.get('.cdk-overlay-container .cdk-overlay-pane').contains(travelPolicyForm.trainTo).click();
+    cy.get('.cdk-overlay-container dib-travel-policy-dialog dib-list-item input[placeholder="Budget per train"]').type(
+      travelPolicyForm.travelPolicyBudget
+    );
+    cy.get('.cdk-overlay-container dib-travel-policy-dialog input[placeholder=Search]').type(
+      travelPolicyForm.employeeName
+    );
+    cy.get('.cdk-overlay-container dib-travel-policy-dialog .members').click();
+    cy.get('.cdk-overlay-container dib-travel-policy-dialog ui-button[type=success]').click();
+    cy.get('dib-company-management dib-travel-policy dib-expandable-item .section__header__title').should(
+      'contain',
+      travelPolicyForm.travelPolicyName
+    );
+  });
+
+  it('should extend form of travel policy and display details', () => {
+    cy.get('dib-company-management dib-travel-policy dib-expandable-item .collapsed .button').first().dblclick();
+  });
+
+  it('updates a train travel policy', () => {
+    cy.get('dib-company-management dib-travel-policy dib-expandable-item .section__header__title')
+      .contains(travelPolicyForm.travelPolicyName)
+      .first()
+      .parents('dib-expandable-item')
+      .within(() => {
+        return cy.get('ui-button').contains('edit').click();
+      });
+    cy.get('.cdk-overlay-container dib-travel-policy-dialog input[placeholder=Name]')
+      .clear()
+      .type(travelPolicyForm.travelPolicyNameUpdate);
+    cy.get('.cdk-overlay-container dib-travel-policy-dialog ui-button[type=success]').click();
+    cy.get('dib-company-management dib-travel-policy dib-expandable-item .section__header__title').should(
+      'contain',
+      travelPolicyForm.travelPolicyNameUpdate
+    );
+  });
+
+  it('deletes an updated train travel policy', () => {
```

Comment:
This test is broken on my side. Please revisit.

```
Timed out retrying after 4050ms: cy.click() failed because this element is detached from the DOM.

<button _ngcontent-xuw-c10="" class="ui-button dib-outline dib-button-warning dib-button-size-sm" ng-reflect-klass="ui-button" ng-reflect-ng-class="[object Object]">delete</button>

Cypress requires elements be attached in the DOM to interact with them.

The previous command that ran was:

  > cy.contains()

This DOM element likely became detached somewhere between the previous and current command.

Common situations why this happens:
  - Your JS framework re-rendered asynchronously
  - Your app code reacted to an event firing and removed the element

You typically need to re-query for the element or add 'guards' which delay Cypress from running new commands.
```